procedures/crates.md

```diff
@@ -0,0 +1,33 @@
+# Crate Policies
+This document describe the guidelines for splitting the compiler up into crates and using
+third-party crates in the compiler. These guidelines were originally discussed at the Rust All
+Hands 2019 by the compiler team and others.
+
+## Where should compiler crates live?
+As library-ification progresses and parts of the compiler are extracted into separate reusable
+crates, these new crates should be hosted in the `rust-lang` organization.
```

Comment:
How do we reach consensus that such a crate/repo should be created? I imagine we want to avoid unnecessary churn where repos are created and then deleted again.

Reply:
cc @rust-lang/libs I think the library team should probably weigh in on this policy. e.g., Whether these crates belong in the `rust-lang` org or not?

Reply:
I would certainly like to hear the @rust-lang/libs team opinion. I don't really view the rust-lang GH org as "belonging" to the libs team (nor to any one team), though.

Reply:
I personally would agree that `rust-lang` is the best place for these crates, and like @nikomatsakis don't necessarily think that libs or any one teams own the org, but rather it's for all our use. We'll develop cruft over time, but that's sort of natural and doesn't tend to detract too much from actively developed projects. We can semi-aggressively archive projects that are no longer used or transfer them to other owners.

Reply:
Sorry, I didn't meant to imply any particular ownership stake here. Just wanted to get more visibility on this. In particular, in the past, crates were only supposed to move into `rust-lang` after going through an RFC process. To me, that implied there was a certain minimum standard of quality given to crates in the `rust-lang` repo. I grant that we don't follow this process any more (with an RFC being too heavyweight and preferring more rapid iteration outside of the `rust-lang` orgs altogether, which I think has been very successful), but it would be good, I think, to be more intentional about changing the messaging here. Namely, that "crates in `rust-lang` are no longer necessarily supposed to be high quality, well vetted core crates in the ecosystem, although some may be."

This kind of raises another question here, which is what determines which crates go into `rust-lang` in the first place. Is it just up to each individual team's discretion? If so, that's fine, it would just be good to have that communicated somewhere. (Or maybe it is and I've missed it.)

Reply:
> Namely, that "crates in rust-lang are no longer necessarily supposed to be high quality, well vetted core crates in the ecosystem, although some may be."

This could be communicated by using GitHub's topics to indicate the team that is responsible for the repository.

Reply:
I've added some clarifications (based on yesterday's steering meeting) that should address @michaelwoerister's initial concerns with this comment. I'm going to resolve the comment so I can keep track of changes I need to make - perhaps the rest of this discussion would be better suited to Zulip or internals.r-l.o?